stackslib/src/chainstate/stacks/db/transactions.rs

```diff
@@ -1334,6 +1351,21 @@ impl StacksChainState {
                 let receipt = StacksTransactionReceipt::from_coinbase(tx.clone());
                 Ok(receipt)
             }
+            TransactionPayload::TenureChange(ref payload) => {
+                // post-conditions are not allowed for this variant, since they're non-sensical.
+                // Their presence in this variant makes the transaction invalid.
+                if tx.post_conditions.len() > 0 {
+                    let msg = format!("Invalid Stacks transaction: TenureChange transactions do not support post-conditions");
+                    warn!("{msg}");
+
+                    return Err(Error::InvalidStacksTransaction(msg, false));
+                }
+
+                // TODO: More checks before adding to block?
```

Comment:
A few:

* This must be the first block in the miner's tenure.  It cannot be in a subsequent block
* This `TenureChange` must be for `last_tenure + 1`.  No `TenureChange`s can be skipped.
* It must point to the last active miner's block-commit's `StacksBlockId`
* It must count up the number of blocks since the last block with a `TenureChange` (it may be zero; a block can have multiple `TenureChange` transactions per the design doc)
* It must be signed by the current Stacker set (this could be checked on `consensus_deserialize`)
* It must precede any other kinds of transactions (this could be checked in `StacksBlock::consensus_deserialize`)

Reply:
I think this should live as a separate issue from this PR and let this PR function mostly just as a wire format implementation (https://github.com/stacks-network/stacks-blockchain/issues/4014)

Reply:
Agreed